test/NuGet.Core.Tests/NuGet.Versioning.Test/VersionRangeTests.cs

```diff
@@ -1503,5 +1509,29 @@ public void FindBestMatch_FloatingPrereleaseMinor_WithExtraDashes()
 
             Assert.Equal("1.9.0--beta", range.FindBestMatch(versions).ToNormalizedString());
         }
+
+        [Theory]
+        [InlineData("[1.1.4, 1.1.2)")]
+        [InlineData("[1.1.4, 1.1.2]")]
+        [InlineData("(1.1.4, 1.1.2)")]
+        [InlineData("(1.1.4, 1.1.2]")]
+        [InlineData("[1.0.0, 1.0.0)")]
+        [InlineData("(1.0.0, 1.0.0]")]
+        [InlineData("(1.0, 1.0.0]")]
+        [InlineData("[*, *)")]
+        [InlineData("(*, *]")]
+        [InlineData("[1.0.0-beta, 1.0.0-beta+900)")]
+        [InlineData("(1.0.0-beta+600, 1.0.0-beta]")]
+        [InlineData("(1.0)")]
+        [InlineData("(1.0.0)")]
+        [InlineData("[2.0.0)")]
+        [InlineData("(2.0.0]")]
+        [InlineData("[2.0.0, *)")]
+        public void VersionRange_LogicallyIncorrectRanges_Throws(string range)
```

Comment:
The correct name according to the guide would be `Parse_****_****`

Reply:
Ok . Renamed.